crates/executor/src/state_reader.rs

```diff
@@ -37,15 +38,15 @@ pub(super) struct PathfinderStateReader<'tx> {
     native_class_cache: Option<NativeClassCache>,
 }
 
-impl<'tx> PathfinderStateReader<'tx> {
+impl PathfinderStateReader {
     pub fn new(
-        transaction: &'tx pathfinder_storage::Transaction<'tx>,
+        db: pathfinder_storage::Storage,
```

Comment:
Until now, transaction execution was done in a single transaction, making sure we're getting a consistent view of storage.

Now each read operation is creating a transaction independently, so basically there's no guarantee that data used by execution does not get pruned by parallel updates.

I don't think this change (execution state capturing a `Storage` instead of a transaction) is safe to do.